### ISSUE TITLE
[Bug]: ResizeObserver is not defined

### ISSUE THREAD

**Issue by melrefaie** [DerYeger/vue-masonry-wall#24]
### Description

Using with nuxt 3 i got this problem. ResizeObserver is not defined
it should support ssr. 
temp fix i use ClientOnly component  to wrap MasonryWall

```
ResizeObserver is not defined
  at setup (file://./.nuxt/dist/server/server.mjs:34622:28)  
  at callWithErrorHandling (./node_modules/@vue/runtime-core/dist/runtime-core.cjs.js:157:22)  
  at setupStatefulComponent (./node_modules/@vue/runtime-core/dist/runtime-core.cjs.js:6977:29)  
  at setupComponent (./node_modules/@vue/runtime-core/dist/runtime-core.cjs.js:6933:11)  
  at renderComponentVNode (./node_modules/@vue/server-renderer/dist/server-renderer.cjs.js:198:17)  
  at Module.ssrRenderComponent (./node_modules/@vue/server-renderer/dist/server-renderer.cjs.js:629:12)  
  at ssrRender (file://./.nuxt/dist/server/server.mjs:28073:31)  
  at renderComponentSubTree (./node_modules/@vue/server-renderer/dist/server-renderer.cjs.js:269:13)  
  at renderComponentVNode (./node_modules/@vue/server-renderer/dist/server-renderer.cjs.js:214:16)  
  at renderVNode (./node_modules/@vue/server-renderer/dist/server-renderer.cjs.js:310:22)
```

### Reproduction

none

### Preferences

- [ ] I want to be assigned to and work on this issue myself

**Comment by DerYeger** [comment]
Hi,

this library does already support SSR, but it requires a simple dummy implementation of `ResizeObserver` during the SSR build. I didn't add it to this library directly to reduce bundle size and keep it simple.

For example, I created `Vite-plugin-ssg-utils` to provide a dummy implementation for `vite-ssg`.
I'm pretty sure Nuxt 3 has similar hooks that are only executed during SSR/SSG builds.

See https://github.com/DerYeger/vite-plugin-ssg-utils/blob/6c82c70cf5f2a08e329be57080aff0b23346b402/src/index.ts#L28 for the dummy.

**Comment by or2e** [comment]
Hi, the Nuxt authors answered: https://github.com/nuxt/framework/discussions/4482
I'll try, but so far it doesn't work

**Comment by DerYeger** [comment]
I just released v3.0.38 that should fix the issue without requiring the above workaround.
